src/Payu/Parser/PaymentResponseParser.php

```diff
@@ -10,23 +11,46 @@
 class PaymentResponseParser implements ParserInterface
 {
     /**
-     * @param string $rawData
+     * @param EntityBody|string|array $rawData
      * @return PaymentResponse|ResponseAbstract
      * @throws \Payu\Exception\BadResponseError
      */
     public function parse($rawData)
     {
         try {
-            $xml = new SimpleXMLElement($rawData);
+            $xml = ($rawData instanceof EntityBody || is_string($rawData)) ?
+                                    new SimpleXMLElement($rawData) : (object) $rawData;
         } catch(Exception $e) {
             throw new BadResponseError('Unexpected response received from provider. Response: ' . $rawData);
         }
+
         $status = (string) $xml->STATUS;
         $code = (string) $xml->RETURN_CODE;
         $message = (string) $xml->RETURN_MESSAGE;
-        $statusCode = $status == 'SUCCESS' && $code == 'AUTHORIZED' ?
-                     ResponseAbstract::STATUS_APPROVED : ResponseAbstract::STATUS_DECLINED;
-        $transactionId = $statusCode == ResponseAbstract::STATUS_APPROVED ? (string) $xml->REFNO : null;
-        return new PaymentResponse($statusCode, $code, $message, $transactionId);
+        $statusCode = $this->getStatusCode($status, $code);
+        $transactionId = $this->getTransactionId($xml->REFNO, $statusCode);
```

Comment:
getTransactionId, getStatusCode gibi metodlar, aslında statuscode veya transactionid yi belirli durumlarda response dan ayrıştırıp dönen metodlar. Eğer parse metodunu daha fazla büyütmemek için böyle bir yaklaşıma gidiyorsan bence parseBlaBla... şeklinde helper metodlar oluştursan daha şık olur. getter metodun argüman alıyor olması biraz kafa karıştırıcı.